### ISSUE TITLE
refactor | backend | #369 collapse retrieve/survey/retrieve_scoped → find_passages

### ISSUE THREAD

**Issue by AmosChenZixuan** [AmosChenZixuan/BibiLab#370]
Part of EPIC #369.

## Why

Current RAG system has three retrieve tools (`retrieve`, `survey`, `retrieve_scoped`) that all collapse to match-and-retrieve under different param tuples (`_PARAMS_BY_MODE`, `_RELEVANCE_MARGIN_BY_MODE`). Mode dispatch is symptomatic tuning, not mechanistic differentiation. Surrounding tech debt accumulated atop this fragmentation:

- Diversity cap (`_diverse_top_k`) pre-emptively dedups across sources, masking clustering signal and starving single-source deep-dive questions
- Neighbor-pull (`is_neighbor` sentinel + scarcity threshold) is a retrieval-time bandaid for chunk-boundary problems already addressed structurally by #359 (pause-aware) and #365 (sentence-aware) chunking
- Defensive facet-strip warnings smell of a tool contract that isn't enforceable in types
- LLM-visible `_NO_MATCH_NOTE` injection has backend writing prompt fragments
- Sequential guard blocks legitimate multi-hop retrieval to prevent waste
- Pre-emptive cross-turn tool-result compaction (`_summarize_stale_retrieve_block`) mostly compensates for keep-raw failure modes that v2's tool design no longer triggers

This issue collapses the three tools into one `find_passages` and removes the tech-debt layer. Pairs with #371 (read_source) and #372 (prompt rewrite) — must land atomically (single PR).

### New tool

`find_passages(query: str, sequence_number?: int, season_number?: int)`

Hybrid retrieval: vector (ChromaDB) ∥ BM25 (FTS5) → RRF fuse (k=60) → cross-encoder rerank (bge-reranker-base) → single fixed-margin gate. Replaces all three current retrieve tools.

Constants in `chat_tools.py`:
- `FIND_PASSAGES_GATE_MARGIN = 2.0` (replaces `_RELEVANCE_MARGIN_BY_MODE`)
- `FIND_PASSAGES_TOP_K = 8` (replaces per-mode top_k)

Facet handling preserved from current `retrieve_scoped`: deterministic facet matching (`sequence_number`, `season_number`) narrows source pool before retrieval. Fail-open: no match → full pool with `facet_scope.no_match=true` in telemetry.

### Output format

Existing `_build_fenced_chunks` per-source fence (post-#297) with `[N]:` chunk markers. Each source fence header carries `sources.summary` (folds #357 in):

```
===== Source [1]: "title"
Summary: {sources.summary}
=====
[1]: "chunk text 1"
[1]: "chunk text 2"

===== Source [2]: "title"
Summary: {sources.summary}
=====
[2]: "chunk text 3"
```

JSON envelope: `{ "_chunks": "<fence string>", "sources_with_hits": int, "candidates_evaluated": int, "gate_margin": float, "facet_scope": {...} }`. Metadata fields are consumed by frontend retrieval ledger (SSE `rag` event); `_client_tool_result()` strips `_chunks` from client payload.

### Deletions

**`embed.py`:**
- `_diverse_top_k` (function + callsite)
- `_adaptive_depth`
- `_PARAMS_BY_MODE`
- `_RELEVANCE_MARGIN_BY_MODE`
- Neighbor block (~80 LOC around line 731 — `neighbor_rows` fetch, `_chunks_from_chroma_get` wrap, scarcity threshold check)
- `RetrievedChunk.is_neighbor` field (struct retained, field removed)
- `_chunks_from_chroma_get` helper
- `_chunk_score` `is_neighbor` +inf branch
- `neighbors_pulled` telemetry counter

**`chat_tools.py`:**
- `retrieve` / `survey` / `retrieve_scoped` tool definitions
- `query_list_metadata` tool definition + `execute_query_list_metadata`
- `generate_report` tool definition (NOTE: `execute_generate_report` is retained — Lab UI still uses it via separate route)
- `_NO_MATCH_NOTE` constant + injection logic in scoped path
- Defensive facet-strip warnings (`_facet_int` retained, strip warning logic removed)
- `_summarize_stale_retrieve_block` + cross-turn tag injection
- chat_tools.py:400 defensive `getattr(c, "is_neighbor", False)` (orphan after field deletion)

**`models/_enums.py`:**
- `RetrievalParams.mode` field

**`config.py`:**
- `rag.neighbor_scarcity_threshold`

**`routers/chat.py`:**
- Sequential guard (active-tools manipulation after first retrieve-family call)
- Stale tool block replay branches for removed tools

**`db.py`:**
- `count_sources` / `longest_source` / `language_breakdown` — grep-verify zero callers at impl time, delete unconditionally if confirmed

### Backward compat with persisted messages

`_summarize_stale_retrieve_block` (the only replay path that branched on tool name) dies in this issue. Surviving load path is `reseed_citation_registry` (`chat_tools.py:719`), which reads emitted citation ledger from message metadata (tool-name agnostic) and rebuilds the registry for `[N]` click-through on historical assistant prose. Legacy `retrieve` / `survey` mentions in stored prose become inert text. No new code required for compat.

## Tests

- Repurpose `test_embed_retrieve` / `test_chat_rag` to cover `find_passages`
- Facet scoping (`sequence_number` / `season_number`) narrows pool correctly
- Fail-open on zero-match facet (full pool returned, telemetry flagged)
- Hybrid → RRF → rerank → gate end-to-end
- Multi-hop allowed (sequential guard removed): two consecutive `find_passages` calls in same turn both succeed

## Acceptance

All tests green. `uv run pytest` clean. Pre-commit hooks (ruff lint/format) green. Run `/simplify` before commit.

Manual smoke (depends on #371 + #372 for full coverage):
- Single-source fact ("when does X happen") → one `find_passages` call returns relevant excerpts
- Scoped fact ("in ep 5 when does X happen") → facet narrows to ep 5
- Multi-hop ("compare X across ep 1 and ep 3") → parallel `find_passages` calls work

**Comment by AmosChenZixuan** [comment]
## Design input: find_passages' "single fixed-margin gate, rerank sole authority" reproduces a real recall failure

Surfaced while fixing #391 (pinyin-expansion OR-join, shipped in `7ad701a`). Recording here because #370's spec — `vector ∥ BM25 → RRF (k=60) → rerank → single fixed-margin gate` with `FIND_PASSAGES_GATE_MARGIN = 2.0` — keeps the exact pipeline that fails below. This is a decision #370 should make deliberately, with evidence.

### The failure
Real query `前美子是谁` ("who is 钱美子"). 钱美子 is the witch's external-brain **fish** (introduced in ep2 as `钱梅子` — ASR homophone of 美子). Ground truth lives in ep2 chunks; the LLM answered "钱美子 = the witch" from an ep3 chunk.

Why: the **cross-encoder reranker reads raw hanzi**, so it's homophone-blind. BM25+pinyin *recall* the ep2 identity chunk fine (#391 fix works), but rerank scores it near-last and the single-margin gate drops it:

```
                       retr_rank  rerank_rank  rerank_score  kept(gate)
b9179385#3 (ep3 女巫)       5           0          -1.36         Y      <- wrong source survives
fffee4a7#5 (ep2 identity)   1           4          -6.01         .
fffee4a7#4 (ep2 identity)   0           7          -7.07         .      <- BM25/pinyin #1, reranked last
gate: threshold = max(median -6.27, top -1.36 - 2.0) = -3.36  →  kept 1 / 10
```

**The gate buried the correct _source_, not just a chunk.** This matters for the v2 escape-hatch story: a "who is X" question is supposed to fall back to `read_source` (#371), but find_passages points the LLM at ep3 (女巫), so `read_source(ep3)` would still answer wrong. **read_source only rescues this if find_passages gets source-level recall right.**

### POCs (throwaway; deduped pool)
- **B — inject pinyin into rerank input** (`hanzi + appended toneless pinyin`): ❌ scores dropped across the board, identity chunks fell further. bge (XLM-R) isn't trained on hanzi+pinyin; romanization is noise. Dead end.
- **C — RRF-fuse retrieval-rank with rerank-rank** (same operator, one level up): ✅ both identity chunks re-enter the kept set (fffee4a7#5→rank1, #4→rank3). Works because retrieval *knows* they're the best name match; rerank loses its sole veto.
- **A — lexical floor** (force-keep top BM25 hits): rejected earlier as too noisy for common words.

### The open decision for #370
1. **Fix source-level recall in find_passages** — stop rerank having sole authority. Two candidates: **C (rank fusion)** or **score interpolation** `α·retr + β·rerank` (more standard for retriever+reranker; preserves rerank magnitude where reliable). Worth a head-to-head; C is a defensible reuse of `_rrf_fuse` but discards magnitude.
2. **OR lean on read_source as the escape hatch** — but then find_passages must *guarantee* the right source appears (even if not the best chunk), which is the same problem in weaker form.

### Gate-domain caveat
Any fusion changes the gate's score domain: RRF-fused scores sit at ~0.03, so `top − 2.0` (bge-logit units) is meaningless — only the median half-cut survives. `FIND_PASSAGES_GATE_MARGIN` needs a rank-based or recalibrated basis if fusion is adopted. (This was the entire premise of the now-closed #392, folded here.)

### Not in scope of this finding
Grounding-prompt patch ("don't merge co-occurring entities") — rejected as a bandaid. Belongs to #372 only if the team wants it; it doesn't fix retrieval.

**Comment by AmosChenZixuan** [comment]
## Scope update (2026-06-01): delete the gate, don't redesign it

Supersedes the fusion/gate-redesign direction in my earlier [design-input comment](https://github.com/AmosChenZixuan/BibiLab/issues/370#issuecomment-4591345011). Razor pass concluded the gate itself is the over-defense.

**Change to §5.2 / §7 / §8 scope:**
- **Delete the relevance gate.** Remove `_quantile_gate`, `_RELEVANCE_MARGIN_BY_MODE`, and `FIND_PASSAGES_GATE_MARGIN` from the plan.
- `find_passages` returns **`FIND_PASSAGES_TOP_K` by rerank order** — rerank is an *ordering*, not an authority. The LLM filters relevance and decides escalation.
- Keep at most a **single absolute zero-coverage floor** (`FIND_PASSAGES_MIN_RELEVANCE`, one constant) so a genuinely-uncovered query returns empty (#332) — or push that to the grounding prompt. Telemetry: `zero_coverage: bool` replaces `gate_margin`.
- `find_passages` is reframed as a **recall-biased locator**, not a precision fact-finder.

**Why:** the median+margin gate collapsed 24 candidates → 1 on `前美子是谁`, hiding the correct source and forcing a confident-wrong answer; rerank score was anti-correlated with correctness on `隐身术的施法三要素`. A cheap locator with a `read_source` backstop + an LLM that filters does not need precision-gating.

**Net:** net-negative LOC, and this **retires #392** (no fused score, no rank fusion / interpolation, no margin recalibration).
